crates/oxc_linter/src/rules/oxc/no_barrel_file.rs

```diff
@@ -60,23 +78,90 @@ impl Rule for NoBarrelFile {
 
         let AstKind::Program(program) = root.kind() else { unreachable!() };
 
-        let declarations =
-            program
-                .body
-                .iter()
-                .fold(0, |acc, node| if node.is_declaration() { acc + 1 } else { acc });
-        let exports =
-            module_record.star_export_entries.len() + module_record.indirect_export_entries.len();
-
-        if exports > declarations
-            && exports > AMOUNT_OF_EXPORTS_TO_CONSIDER_MODULE_AS_BARREL as usize
-        {
-            let loaded_modules_count = ModuleGraphVisitorBuilder::default()
-                .visit_fold(0, module_record, |acc, _, _| VisitFoldWhile::Next(acc + 1))
-                .result;
-            ctx.diagnostic(NoBarrelFileDiagnostic(program.span, loaded_modules_count));
+        if program.body.iter().all(|node| {
+            matches! {
+                node,
+                match_module_declaration!(Statement) if !node.to_module_declaration().is_type()
+            }
+        }) {
+            let diag = if let Some(count) = count_loaded_modules(module_record) {
+                NoBarrelFileDiagnostic::BarrelFileWithDetails(program.span, count)
+            } else {
+                NoBarrelFileDiagnostic::BarrelFile(program.span)
+            };
+
+            ctx.diagnostic(diag);
         }
     }
+
+    fn run(&self, node: &AstNode<'_>, ctx: &LintContext<'_>) {
+        let AstKind::ImportDeclaration(import) = node.kind() else { return };
+        if is_facade_import(ctx.file_path().to_str().unwrap(), import.source.value.as_str()) {
```

Comment:
I can simply this, leave this to me.

Reply:
Feel free to take over this PR, I'll start working on other stuff. Let me know if there is anything you want me to change.